snap/snapcraft.yaml

```diff
@@ -20,82 +18,109 @@ architectures:
-      #    ;;
-      esac
-      # download the archive, failing on ssl cert problems
-      curl https://dl.google.com/go/$FILE_NAME -O
-      echo "$FILE_HASH $FILE_NAME" > sha256
-      sha256sum -c sha256 | grep OK
-      tar -C $SNAPCRAFT_STAGE -xf go*.tar.gz --strip-components=1
-    prime:
-      - "-*"
-
-  device-rfid-llrp-go:
+      cd $SNAPCRAFT_PART_SRC
+      make build
+      install -DT ./cmd/configure/configure "$SNAPCRAFT_PART_INSTALL/snap/hooks/configure"
+      install -DT ./cmd/install/install "$SNAPCRAFT_PART_INSTALL/snap/hooks/install"
+
+  device-rfid-llrp:
     source: .
     plugin: make
-    build-packages: [git]
-    after: [go]
+    build-packages: [git, libzmq3-dev, pkg-config]
+    stage-packages: [libzmq5]
+    build-snaps:
+      - go/1.16/stable
     override-build: |
       cd $SNAPCRAFT_PART_SRC
+
+      # The makefile uses a VERSION file to set the Version string. 
+      # This file needs to be generated here
+      GIT_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+      if [ -z "$GIT_VERSION" ]; then
+        GIT_VERSION="0.0.0"
+      fi
+      echo $GIT_VERSION > ./VERSION
+
+      go mod tidy
       make build
 
-      install -DT "./cmd/device-rfid-llrp-go" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp-go" 
+      install -DT "./cmd/device-rfid-llrp" "$SNAPCRAFT_PART_INSTALL/bin/device-rfid-llrp"
 
-      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp-go/res/"
-      
-      cat "./cmd/res/configuration.toml" | \
-        sed -e s:\"./device-rfid-llrp-go.log\":\'\$SNAP_COMMON/device-rfid-llrp-go.log\': \
-          -e s:'ProfilesDir = \"./res\"':'ProfilesDir = \"\$SNAP_DATA/config/device-rfid-llrp-go/res\"': > \
-        "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp-go/res/configuration.toml"
+      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/"
+      install -DT "./cmd/res/configuration.toml" \
+        "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/configuration.toml"
 
-      if [ "$(find ./cmd/res/ -maxdepth 1 -name *.yaml -print -quit)" ]; then  
-       for fpath in ./cmd/res/*.yaml; do
+      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/profiles/"
+      for fpath in ./cmd/res/profiles/*.yaml; do
           fname=$(basename "$fpath")
-          install -DT "./cmd/res/${fname}" \
-            "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp-go/res/${fname}"
-       done
-      fi
-   
+          install -DT "./cmd/res/profiles/${fname}" \
+            "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/profiles/${fname}"
+      done
+
+      install -d "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/devices/"
+      for fpath in ./cmd/res/devices/*.toml; do
+          fname=$(basename "$fpath")
+          install -DT "./cmd/res/devices/${fname}" \
+            "$SNAPCRAFT_PART_INSTALL/config/device-rfid-llrp/res/devices/${fname}"
+      done
```

Comment:
What is being copied here don't exist in the source files.

Reply:
I have removed those lines